sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -770,12 +770,10 @@ final void moveToOutputFiles(
       }
       // During a failure case, files may have been deleted in an earlier step. Thus
       // we ignore missing files here.
-      FileSystems.rename(
-          srcFiles,
-          dstFiles,
-          StandardMoveOptions.IGNORE_MISSING_FILES,
-          StandardMoveOptions.SKIP_IF_DESTINATION_EXISTS);
-      removeTemporaryFiles(srcFiles);
```

Comment:
I think we still need removeTemporaryFiles() to remove the temporary directory for batch.

Reply:
adding back, I think we need this as well if we are filtering if the destination exists because we still need to remove the source in that case.

Reply:
Might make sense to update the semantics of  "StandardMoveOptions.SKIP_IF_DESTINATION_EXISTS" to delete the source during rename. That will allow us to prevent the double delete for the case where the source existed.

Also probably removeTemporaryFiles() should be updated to just cleanup the temporary directory (where appropriate) instead of trying to delete already renamed files.